gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-http-proxy/src/main/java/io/gravitee/plugin/endpoint/http/proxy/HttpProxyEndpointConnectorFactory.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.plugin.endpoint.http.proxy;
+
+import io.gravitee.common.http.HttpHeader;
+import io.gravitee.definition.model.v4.http.HttpProxyOptions;
+import io.gravitee.el.TemplateEngine;
+import io.gravitee.gateway.jupiter.api.ConnectorMode;
+import io.gravitee.gateway.jupiter.api.connector.ConnectorHelper;
+import io.gravitee.gateway.jupiter.api.connector.endpoint.sync.EndpointSyncConnectorFactory;
+import io.gravitee.gateway.jupiter.api.context.DeploymentContext;
+import io.gravitee.gateway.jupiter.api.exception.PluginConfigurationException;
+import io.gravitee.gateway.jupiter.reactor.handler.context.DefaultDeploymentContext;
+import io.gravitee.plugin.endpoint.http.proxy.configuration.HttpProxyEndpointConnectorConfiguration;
+import java.util.List;
+import java.util.Set;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@AllArgsConstructor
+public class HttpProxyEndpointConnectorFactory implements EndpointSyncConnectorFactory<HttpProxyEndpointConnector> {
+
+    private final ConnectorHelper connectorFactoryHelper;
+
+    @Override
+    public Set<ConnectorMode> supportedModes() {
+        return HttpProxyEndpointConnector.SUPPORTED_MODES;
+    }
+
+    public HttpProxyEndpointConnector createConnector(final DeploymentContext deploymentContext, final String configuration) {
+        try {
+            return new HttpProxyEndpointConnector(
+                eval(
+                    deploymentContext,
+                    connectorFactoryHelper.readConfiguration(HttpProxyEndpointConnectorConfiguration.class, configuration)
+                )
+            );
+        } catch (Exception e) {
+            log.error("Can't create connector cause no valid configuration", e);
+            return null;
+        }
+    }
+
+    private HttpProxyEndpointConnectorConfiguration eval(
```

Comment:
This is probably something we should introduce on other endpoint.

Reply:
No sure if you request any changes here or if it just a thought 🧐

Reply:
a thought for other endpoint

Reply:
Can you be more specific cause I don't get it sorry. I don't have any other example of endpoint which support el on some particular fields. The fields supporting EL here are really specific to http so I don't see how this part can be shared tbh.